src/sagemaker/jumpstart/artifacts/hyperparameters.py

```diff
@@ -86,4 +87,19 @@ def _retrieve_default_hyperparameters(
             include_container_hyperparameters and hyperparameter.scope == VariableScope.CONTAINER
         ) or hyperparameter.scope == VariableScope.ALGORITHM:
             default_hyperparameters[hyperparameter.name] = str(hyperparameter.default)
+
+    instance_specific_hyperparameters = (
+        model_specs.training_instance_type_variants.get_instance_specific_hyperparameters(
+            instance_type
+        )
+        if instance_type
+        and getattr(model_specs, "training_instance_type_variants", None) is not None
+        else []
+    )
+
+    for instance_specific_hyperparameter in instance_specific_hyperparameters:
+        default_hyperparameters[instance_specific_hyperparameter.name] = str(
```

Comment:
Would we ever want to remove a hyperparameter for a specific instance? If so, overriding the defaults may not work.

In the generated metadata, shouldn't the instance-specific hyperparameter dict contain all the fields rather than overriden ones?

Reply:
That's a good point. However, to keep consistency with the environment variable variants, I think it should be additive. 
In the situation you describe, the instance family or default metadata should be scoped down so that no hyperparameters need to be removed. In other words, we can construct any arbitrary hyperparameters  for instances using this metadata design.